src/main/scala/com/advancedtelematic/director/manifest/DeviceManifestUpdate.scala

```diff
@@ -1,28 +1,47 @@
 package com.advancedtelematic.director.manifest
 
+import java.security.MessageDigest
+
+import akka.http.scaladsl.util.FastFuture
 import cats.syntax.show._
 import com.advancedtelematic.director.data.Codecs._
 import com.advancedtelematic.director.data.DeviceRequest.{CustomManifest, EcuManifest}
-import com.advancedtelematic.director.db.{DeviceRepositorySupport, DeviceUpdate, DeviceUpdateResult, UpdateTypesRepositorySupport}
+import com.advancedtelematic.director.db.{DeviceRepositorySupport, DeviceUpdate, DeviceUpdateResult, ProcessedManifestsRepositorySupport, UpdateTypesRepositorySupport}
 import com.advancedtelematic.director.manifest.Verifier.Verifier
 import com.advancedtelematic.libats.data.DataType.Namespace
 import com.advancedtelematic.libats.messaging_datatype.DataType.{DeviceId, EcuSerial}
-import com.advancedtelematic.libtuf.data.TufDataType.{JsonSignedPayload, OperationResult, SignedPayload, TufKey}
+import com.advancedtelematic.libtuf.data.TufDataType.{OperationResult, SignedPayload, TufKey}
 import io.circe.Json
 import org.slf4j.LoggerFactory
 
 import scala.async.Async._
 import scala.concurrent.{ExecutionContext, Future}
 import slick.driver.MySQLDriver.api._
+import com.advancedtelematic.libtuf.crypt.CanonicalJson._
 
 class DeviceManifestUpdate(afterUpdate: AfterDeviceManifestUpdate,
                            verifier: TufKey => Verifier
                           )(implicit val db: Database, val ec: ExecutionContext)
-    extends DeviceRepositorySupport
-    with UpdateTypesRepositorySupport {
+    extends DeviceRepositorySupport with UpdateTypesRepositorySupport with ProcessedManifestsRepositorySupport {
   private lazy val _log = LoggerFactory.getLogger(this.getClass)
+  val digester: MessageDigest = MessageDigest.getInstance("SHA-256")
+
+  def setDeviceManifest(namespace: Namespace, device: DeviceId, signedDevMan: SignedPayload[Json]): Future[Unit] = {
+    val digest = digester.digest(signedDevMan.signed.canonical.getBytes("UTF-8"))
```

Comment:
You can use `com.advancedtelematic.libtuf_server.crypto.Sha256Digest.digest().hash.get` 

But this is a bit finicky, if a space gets added to the manifest or something then it will be processed again

Reply:
I think the chance that the manifest format changes are low and it also won't break things.

Reply:
OTOH what could happen is that `timeserver_time`s get corruptly filled in and change all of a sudden?